src/main/java/org/openpnp/gui/JobPanel.java

```diff
@@ -1001,9 +1002,38 @@ public void populatePanelSettingsIntoBoardLocations() {
 
         @Override
         public void actionPerformed(ActionEvent arg0) {
-            UiUtils.messageBoxOnException(() -> {
-                fsm.send(Message.StartOrPause);
-            });
+        	Boolean allPlaced = true;
+        	
+        	boardloop:
```

Comment:
Please break this out into a function - something like `boolean isAllPlaced()`, or you could make it checkAllPlaced and have it perform the prompt as well. As this code stands now it buries the actual intent of the action (start the job) with the check.